app/src/main/java/com/android/sample/ui/swipePage/SwipePage.kt

```diff
@@ -1,6 +1,7 @@
 package com.android.sample.ui.swipePage
 
 import android.annotation.SuppressLint
+import android.util.Log
```

Comment:
LGTM!

1. **`checkIfFilterUsed` Function:**  
   Nice addition! Centralizing the logic to determine if filters are active simplifies the code and reduces redundancy.

2. **`FilterChip` Updates:**  
   Adding the `checkIfFilterUsed` call within `onDelete` actions is a good improvement. It ensures filters are reset only when needed and avoids unnecessary updates.

3. **Code Cleanup:**  
   Using `checkIfFilterUsed` for all filter-related chips (time, difficulty, category) creates consistency and improves readability.